--- origsrc/codeblocks-10.05-release/acinclude.m4	2011-10-30 08:00:58.113614600 -0500
+++ src/codeblocks-10.05-release/acinclude.m4	2011-10-30 03:01:07.784753400 -0500
@@ -36,7 +36,12 @@ AC_DEFUN([CODEBLOCKS_GET_PLATFORM],
 
 AC_DEFUN([CODEBLOCKS_SETUP_FOR_TARGET],
 [case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin*)
+	AC_SUBST(SHARED_FLAGS, "-shared -no-undefined -Xlinker --export-all-symbols")
+	AC_SUBST(PLUGIN_FLAGS, "-shared -no-undefined -avoid-version")
+	linux=true
+;;
+*-*-mingw*)
 	AC_SUBST(SHARED_FLAGS, "-shared -no-undefined -Xlinker --export-all-symbols")
 	AC_SUBST(PLUGIN_FLAGS, "-shared -no-undefined -avoid-version")
 	AC_CHECK_TOOL(RC, windres)
--- origsrc/codeblocks-10.05-release/src/build_tools/autorevision/Makefile.am	2011-10-30 08:00:58.160414700 -0500
+++ src/codeblocks-10.05-release/src/build_tools/autorevision/Makefile.am	2011-10-30 03:01:07.000000000 -0500
@@ -5,5 +5,5 @@ INCLUDES = -I$(top_srcdir)/src/include
 
 auto_revision_LDADD = ../../base/tinyxml/libtinyxml.la
 
-all-local: auto_revision
+all-local: auto_revision$(EXEEXT)
 	./auto_revision +int +wx +t $(top_srcdir) $(top_srcdir)/src/include/autorevision.h
--- origsrc/codeblocks-10.05-release/src/include/prep.h	2011-10-30 08:00:58.176014800 -0500
+++ src/codeblocks-10.05-release/src/include/prep.h	2011-10-30 03:01:07.000000000 -0500
@@ -139,6 +139,7 @@ namespace platform
         platform_freebsd,
         platform_netbsd,
         platform_openbsd,
+        platform_cygwin,
         platform_darwin,
         platform_solaris,
         platform_macosx
@@ -163,6 +164,8 @@ namespace platform
     const identifier id = platform_netbsd;
     #elif defined ( OPENBSD )    || defined ( __OPENBSD__ )
     const identifier id = platform_openbsd;
+    #elif defined ( __CYGWIN__ )
+    const identifier id = platform_cygwin;
     #elif defined ( DARWIN )     || defined ( __APPLE__ )
     const identifier id = platform_darwin;
     #elif defined(sun) || defined(__sun)
@@ -203,9 +206,10 @@ namespace platform
     const bool freebsd = (id == platform_freebsd);
     const bool netbsd  = (id == platform_netbsd);
     const bool openbsd = (id == platform_openbsd);
+    const bool cygwin  = (id == platform_cygwin);
     const bool darwin  = (id == platform_darwin);
     const bool solaris = (id == platform_solaris);
-    const bool unix    = (linux | freebsd | netbsd | openbsd | darwin | solaris);
+    const bool unix    = (linux | freebsd | netbsd | openbsd | cygwin | darwin | solaris);
 
     const int bits = 8*sizeof(void*);
 
--- origsrc/codeblocks-10.05-release/src/plugins/contrib/MouseSap/Makefile.am	2010-05-22 05:20:19.000000000 -0500
+++ src/codeblocks-10.05-release/src/plugins/contrib/MouseSap/Makefile.am	2011-10-30 04:51:15.471952600 -0500
@@ -9,7 +9,7 @@ pluginlib_LTLIBRARIES = libMouseSap.la
 
 libMouseSap_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libMouseSap_la_LIBADD = $(PLUGIN_WX_LIBS) ../../../sdk/libcodeblocks.la
+libMouseSap_la_LIBADD = $(WX_LIBS) $(WX_GTK2_LIBS) ../../../sdk/libcodeblocks.la
 
 libMouseSap_la_SOURCES = MouseSap.cpp
 
--- origsrc/codeblocks-10.05-release/src/plugins/contrib/wxSmith/Makefile.am	2011-10-30 08:00:58.238414900 -0500
+++ src/codeblocks-10.05-release/src/plugins/contrib/wxSmith/Makefile.am	2011-10-30 03:01:07.000000000 -0500
@@ -14,7 +14,7 @@ INCLUDES = $(WX_CXXFLAGS) \
 
 lib_LTLIBRARIES = libwxsmithlib.la
 
-libwxsmithlib_la_LDFLAGS = -version-info 0:1:0 -shared
+libwxsmithlib_la_LDFLAGS = -version-info 0:1:0 -shared -no-undefined
 
 libwxsmithlib_la_LIBADD = $(PLUGIN_WX_LIBS) \
 				../../../sdk/libcodeblocks.la \
--- origsrc/codeblocks-10.05-release/src/plugins/contrib/wxSmithContribItems/wxchart/wxchart-1.0/Makefile.am	2010-05-22 05:17:06.000000000 -0500
+++ src/codeblocks-10.05-release/src/plugins/contrib/wxSmithContribItems/wxchart/wxchart-1.0/Makefile.am	2011-10-30 04:40:37.925082000 -0500
@@ -9,7 +9,7 @@ wxsmithcontribitemsdir=$(libdir)/wxSmith
 
 wxsmithcontribitems_LTLIBRARIES = libwxchartctrl.la
 
-libwxchartctrl_la_LDFLAGS = -version-info 0:1:0
+libwxchartctrl_la_LDFLAGS = -version-info 0:1:0 -no-undefined
 
 libwxchartctrl_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-10.05-release/src/plugins/contrib/wxSmithContribItems/wxflatnotebook/wxFlatNotebook/Makefile.am	2010-05-22 05:17:34.000000000 -0500
+++ src/codeblocks-10.05-release/src/plugins/contrib/wxSmithContribItems/wxflatnotebook/wxFlatNotebook/Makefile.am	2011-10-30 04:40:09.782632900 -0500
@@ -8,7 +8,7 @@ wxsmithcontribitemsdir=$(libdir)/wxSmith
 
 wxsmithcontribitems_LTLIBRARIES = libwxflatnotebook.la
 
-libwxflatnotebook_la_LDFLAGS = -version-info 0:1:0
+libwxflatnotebook_la_LDFLAGS = -version-info 0:1:0 -no-undefined
 
 libwxflatnotebook_la_LIBADD = $(WX_LIBS)
 
@@ -35,4 +35,4 @@ noinst_HEADERS = \
 ./include/wx/wxFlatNotebook/fnb_customize_dlg.h \
 ./include/wx/wxFlatNotebook/fnb_resources.h \
 ./include/wx/wxFlatNotebook/popup_dlg.h \
-./include/wx/wxFlatNotebook/xh_fnb.h
\ No newline at end of file
+./include/wx/wxFlatNotebook/xh_fnb.h
--- origsrc/codeblocks-10.05-release/src/plugins/contrib/wxSmithContribItems/wxthings/wxthings/Makefile.am	2010-05-22 05:18:32.000000000 -0500
+++ src/codeblocks-10.05-release/src/plugins/contrib/wxSmithContribItems/wxthings/wxthings/Makefile.am	2011-10-30 04:40:55.537512700 -0500
@@ -8,7 +8,7 @@ wxsmithcontribitemsdir=$(libdir)/wxSmith
 
 wxsmithcontribitems_LTLIBRARIES = libwxcustombutton.la
 
-libwxcustombutton_la_LDFLAGS = -version-info 0:1:0
+libwxcustombutton_la_LDFLAGS = -version-info 0:1:0 -no-undefined
 
 libwxcustombutton_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-10.05-release/src/sdk/Makefile.am	2011-10-30 08:00:58.285214900 -0500
+++ src/codeblocks-10.05-release/src/sdk/Makefile.am	2011-10-30 03:01:07.000000000 -0500
@@ -10,7 +10,7 @@ INCLUDES = $(WX_CXXFLAGS) \
 
 lib_LTLIBRARIES = libcodeblocks.la
 
-libcodeblocks_la_LDFLAGS = -version-info @CODEBLOCKS_VERSION_INFO@
+libcodeblocks_la_LDFLAGS = -version-info @CODEBLOCKS_VERSION_INFO@ -no-undefined
 
 libcodeblocks_la_LIBADD =  $(WX_LIBS) \
 			scripting/bindings/libsqbindings.la \
--- origsrc/codeblocks-10.05-release/src/sdk/pluginmanager.cpp	2011-10-30 08:00:58.300815000 -0500
+++ src/codeblocks-10.05-release/src/sdk/pluginmanager.cpp	2011-10-30 03:01:07.000000000 -0500
@@ -276,11 +276,11 @@ bool PluginManager::InstallPlugin(const
     wxString resourceName = basename + _T(".zip");
     wxString settingsOnName = basename + _T(".png");
     wxString settingsOffName = basename + _T("-off.png");
-    if (!platform::windows && resourceName.StartsWith(_T("lib")))
+    if (!platform::windows && resourceName.StartsWith(_T("cyg")))
         resourceName.Remove(0, 3);
-    if (!platform::windows && settingsOnName.StartsWith(_T("lib")))
+    if (!platform::windows && settingsOnName.StartsWith(_T("cyg")))
         settingsOnName.Remove(0, 3);
-    if (!platform::windows && settingsOffName.StartsWith(_T("lib")))
+    if (!platform::windows && settingsOffName.StartsWith(_T("cyg")))
         settingsOffName.Remove(0, 3);
     wxString pluginFilename = pluginDir + _T('/') + localName;
 //    Manager::Get()->GetLogManager()->DebugLog(F(_T("Plugin filename: ") + pluginFilename));
@@ -382,11 +382,11 @@ bool PluginManager::UninstallPlugin(cbPl
             resourceFilename = fname.GetName() + _T(".zip");
             settingsOnFilename = fname.GetName() + _T(".png");
             settingsOffFilename = fname.GetName() + _T("-off.png");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
-            if (!platform::windows && settingsOnFilename.StartsWith(_T("lib")))
+            if (!platform::windows && settingsOnFilename.StartsWith(_T("cyg")))
                 settingsOnFilename.Remove(0, 3);
-            if (!platform::windows && settingsOffFilename.StartsWith(_T("lib")))
+            if (!platform::windows && settingsOffFilename.StartsWith(_T("cyg")))
                 settingsOffFilename.Remove(0, 3);
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
             settingsOnFilename = ConfigManager::LocateDataFile(_T("images/settings/") + settingsOnFilename, sdDataGlobal | sdDataUser);
@@ -517,14 +517,14 @@ bool PluginManager::ExportPlugin(cbPlugi
 
             // now get the resource zip filename
             resourceFilename = fname.GetName() + _T(".zip");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
             sourcefiles.Add(resourceFilename);
 
             // the highlighted icon the plugin may have for its "settings" page
             resourceFilename = fname.GetName() + _T(".png");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             resourceFilename.Prepend(_T("images/settings/"));
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
@@ -533,7 +533,7 @@ bool PluginManager::ExportPlugin(cbPlugi
 
             // the non-highlighted icon the plugin may have for its "settings" page
             resourceFilename = fname.GetName() + _T("-off.png");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             resourceFilename.Prepend(_T("images/settings/"));
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
@@ -542,7 +542,7 @@ bool PluginManager::ExportPlugin(cbPlugi
 
             // export extra files
             resourceFilename = fname.GetName() + _T(".zip");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             ReadExtraFilesFromManifestFile(resourceFilename, extrafilesdest);
             for (size_t n = 0; n < extrafilesdest.GetCount(); ++n)
@@ -730,7 +730,7 @@ bool PluginManager::ReadManifestFile(con
         wxString actual = fname.GetFullName();
 
         // remove 'lib' prefix from plugin name (if any)
-        if (!platform::windows && actual.StartsWith(_T("lib")))
+        if (!platform::windows && actual.StartsWith(_T("cyg")))
             actual.Remove(0, 3);
 
         actual = ConfigManager::LocateDataFile(actual, sdPluginsUser | sdDataUser | sdPluginsGlobal | sdDataGlobal);
@@ -870,7 +870,7 @@ void PluginManager::ReadExtraFilesFromMa
     wxString actual = fname.GetFullName();
 
     // remove 'lib' prefix from plugin name (if any)
-    if (!platform::windows && actual.StartsWith(_T("lib")))
+    if (!platform::windows && actual.StartsWith(_T("cyg")))
         actual.Remove(0, 3);
 
     actual = ConfigManager::LocateDataFile(actual, sdPluginsUser | sdDataUser | sdPluginsGlobal | sdDataGlobal);
@@ -928,7 +928,7 @@ void PluginManager::ReadExtraFilesFromMa
 
 int PluginManager::ScanForPlugins(const wxString& path)
 {
-    static const wxString PluginsMask = platform::windows ? _T("*.dll") : _T("*.so");
+    static const wxString PluginsMask = _T("*.dll");
     int count = 0;
     if(!wxDirExists(path))
         return count;
@@ -946,10 +946,7 @@ int PluginManager::ScanForPlugins(const
         if (!bbplugins.GetCount())
         {
             // defaults
-            if(platform::windows)
-                bbplugins.Add(_T("compiler.dll"));
-            else
-                bbplugins.Add(_T("libcompiler.so"));
+            bbplugins.Add(_T("cygcompiler.dll"));
         }
     }
 
--- origsrc/codeblocks-10.05-release/src/src/compilersettingsdlg.cpp	2011-10-30 08:00:58.347615100 -0500
+++ src/codeblocks-10.05-release/src/src/compilersettingsdlg.cpp	2011-10-30 03:01:07.000000000 -0500
@@ -66,10 +66,7 @@ CompilerSettingsDlg::CompilerSettingsDlg
     if (!bbplugins.GetCount())
     {
         // defaults
-        if(platform::windows)
-            bbplugins.Add(_T("compiler.dll"));
-        else
-            bbplugins.Add(_T("libcompiler.so"));
+        bbplugins.Add(_T("cygcompiler.dll"));
     }
     wxCheckListBox* clb = XRCCTRL(*this, "chkBBPlugins", wxCheckListBox);
     clb->Clear();
@@ -261,7 +258,7 @@ void CompilerSettingsDlg::EndModal(int r
             }
         }
 
-        const wxString compiler(platform::windows ? _T("compiler.dll") : _T("libcompiler.so"));
+        const wxString compiler(_T("cygcompiler.dll"));
 
         if (bbplugins.Index(compiler) == wxNOT_FOUND)
         {
