--- origsrc/codeblocks-12.11release8629/acinclude.m4	2013-10-09 19:19:13.079323400 -0500
+++ src/codeblocks-12.11release8629/acinclude.m4	2013-10-09 14:06:00.000000000 -0500
@@ -36,7 +36,12 @@ AC_DEFUN([CODEBLOCKS_GET_PLATFORM],
 
 AC_DEFUN([CODEBLOCKS_SETUP_FOR_TARGET],
 [case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin*)
+	AC_SUBST(SHARED_FLAGS, "-shared -no-undefined -Xlinker --export-all-symbols")
+	AC_SUBST(PLUGIN_FLAGS, "-shared -no-undefined -avoid-version")
+	linux=true
+;;
+*-*-mingw*)
 	AC_SUBST(SHARED_FLAGS, "-shared -no-undefined -Xlinker --export-all-symbols")
 	AC_SUBST(PLUGIN_FLAGS, "-shared -no-undefined -avoid-version")
 	AC_CHECK_TOOL(RC, windres)
--- origsrc/codeblocks-12.11release8629/src/build_tools/autorevision/Makefile.am	2013-10-09 19:19:13.083323600 -0500
+++ src/codeblocks-12.11release8629/src/build_tools/autorevision/Makefile.am	2013-10-09 14:06:00.000000000 -0500
@@ -5,5 +5,5 @@ INCLUDES = -I$(top_srcdir)/src/include
 
 auto_revision_LDADD = ../../base/tinyxml/libtinyxml.la
 
-all-local: auto_revision
+all-local: auto_revision$(EXEEXT)
 	./auto_revision +int +wx +t $(top_srcdir) $(top_builddir)/src/include/autorevision.h
--- origsrc/codeblocks-12.11release8629/src/include/prep.h	2013-10-09 19:19:13.086323800 -0500
+++ src/codeblocks-12.11release8629/src/include/prep.h	2013-10-09 14:06:00.000000000 -0500
@@ -158,6 +158,7 @@ namespace platform
         platform_freebsd,
         platform_netbsd,
         platform_openbsd,
+        platform_cygwin,
         platform_darwin,
         platform_solaris,
         platform_macosx
@@ -182,6 +183,8 @@ namespace platform
     const identifier id = platform_netbsd;
     #elif defined ( OPENBSD )    || defined ( __OPENBSD__ )
     const identifier id = platform_openbsd;
+    #elif defined ( __CYGWIN__ )
+    const identifier id = platform_cygwin;
     #elif defined ( DARWIN )     || defined ( __APPLE__ )
     const identifier id = platform_darwin;
     #elif defined(sun) || defined(__sun)
@@ -222,9 +225,10 @@ namespace platform
     const bool freebsd = (id == platform_freebsd);
     const bool netbsd  = (id == platform_netbsd);
     const bool openbsd = (id == platform_openbsd);
+    const bool cygwin  = (id == platform_cygwin);
     const bool darwin  = (id == platform_darwin);
     const bool solaris = (id == platform_solaris);
-    const bool unix    = (linux | freebsd | netbsd | openbsd | darwin | solaris);
+    const bool unix    = (linux | freebsd | netbsd | openbsd | cygwin | darwin | solaris);
 
     const int bits = 8*sizeof(void*);
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/MouseSap/Makefile.am	2013-10-09 19:19:13.091324100 -0500
+++ src/codeblocks-12.11release8629/src/plugins/contrib/MouseSap/Makefile.am	2013-10-09 14:06:00.000000000 -0500
@@ -9,7 +9,7 @@ pluginlib_LTLIBRARIES = libMouseSap.la
 
 libMouseSap_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
-libMouseSap_la_LIBADD = $(PLUGIN_WX_LIBS) ../../../sdk/libcodeblocks.la
+libMouseSap_la_LIBADD = $(WX_LIBS) $(WX_GTK2_LIBS) ../../../sdk/libcodeblocks.la
 
 libMouseSap_la_SOURCES = MouseSap.cpp
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/SmartIndent/manifest.xml	1969-12-31 18:00:00.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/SmartIndent/manifest.xml	2013-10-09 15:38:32.735019000 -0500
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
+<CodeBlocks_plugin_manifest_file>
+    <SdkVersion major="1" minor="11" release="12" />
+    <Plugin name="SmartIndentXML">
+        <Value title="SmartIndentXML" />
+        <Value version="0.1" />
+        <Value description="Provides Smart indenting for XML and HTML" />
+        <Value author="Alpha" />
+        <Value authorEmail="" />
+        <Value authorWebsite="" />
+        <Value thanksTo="" />
+        <Value license="GPL" />
+    </Plugin>
+</CodeBlocks_plugin_manifest_file>
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/source_exporter/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/source_exporter/Makefile.am	2013-10-09 15:36:21.216496600 -0500
@@ -15,7 +15,7 @@ libexporter_la_CPPFLAGS = -DwxPDF_USE_WX
 libexporter_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
 libexporter_la_LIBADD = $(PLUGIN_WX_LIBS) ../../../sdk/libcodeblocks.la \
-				wxPdfDocument/libwxPdfDocument.la
+				wxPdfDocument/libwxPdfDocument.la -lfontconfig
 
 libexporter_la_SOURCES = BaseExporter.cpp \
 				exporter.cpp \
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/KWIC/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/KWIC/Makefile.am	2013-10-09 16:15:09.994695100 -0500
@@ -7,7 +7,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwxkwic.la
 
-libwxkwic_la_LDFLAGS = -version-info 0:1:0
+libwxkwic_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwxkwic_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxFlatNotebook/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxFlatNotebook/Makefile.am	2013-10-09 16:15:10.007695800 -0500
@@ -5,7 +5,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwxflatnotebook.la
 
-libwxflatnotebook_la_LDFLAGS = -version-info 0:1:0
+libwxflatnotebook_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwxflatnotebook_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxImagePanel/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxImagePanel/Makefile.am	2013-10-09 16:15:10.018696500 -0500
@@ -5,7 +5,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwximagepanel.la
 
-libwximagepanel_la_LDFLAGS = -version-info 0:1:0
+libwximagepanel_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwximagepanel_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxSpeedButton/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxSpeedButton/Makefile.am	2013-10-09 16:15:10.029697100 -0500
@@ -5,7 +5,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwxspeedbutton.la
 
-libwxspeedbutton_la_LDFLAGS = -version-info 0:1:0
+libwxspeedbutton_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwxspeedbutton_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxTreeList/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxTreeList/Makefile.am	2013-10-09 16:15:10.042697800 -0500
@@ -5,7 +5,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwxtreelist.la
 
-libwxtreelist_la_LDFLAGS = -version-info 0:1:0
+libwxtreelist_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwxtreelist_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxchart/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxchart/Makefile.am	2013-10-09 16:15:10.055698600 -0500
@@ -6,7 +6,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwxchartctrl.la
 
-libwxchartctrl_la_LDFLAGS = -version-info 0:1:0
+libwxchartctrl_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwxchartctrl_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxled/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxled/Makefile.am	2013-10-09 16:15:10.068699300 -0500
@@ -10,7 +10,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwxled.la
 
-libwxled_la_LDFLAGS = -version-info 0:1:0
+libwxled_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwxled_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxthings/Makefile.am	2012-11-09 08:47:06.000000000 -0600
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxContribItems/wxthings/Makefile.am	2013-10-09 16:15:10.080700000 -0500
@@ -5,7 +5,7 @@ wxcontribitemsdir=$(libdir)/codeblocks/w
 
 wxcontribitems_LTLIBRARIES = libwxcustombutton.la
 
-libwxcustombutton_la_LDFLAGS = -version-info 0:1:0
+libwxcustombutton_la_LDFLAGS = -version-info 0:1:0 -release $(PACKAGE) -no-undefined
 
 libwxcustombutton_la_LIBADD = $(WX_LIBS)
 
--- origsrc/codeblocks-12.11release8629/src/plugins/contrib/wxSmith/Makefile.am	2013-10-09 19:19:13.094324300 -0500
+++ src/codeblocks-12.11release8629/src/plugins/contrib/wxSmith/Makefile.am	2013-10-09 14:06:00.000000000 -0500
@@ -12,7 +12,7 @@ INCLUDES = $(WX_CXXFLAGS) \
 
 lib_LTLIBRARIES = libwxsmithlib.la
 
-libwxsmithlib_la_LDFLAGS = -version-info 0:1:0 -shared
+libwxsmithlib_la_LDFLAGS = -version-info 0:1:0 -shared -release $(PACKAGE) -no-undefined
 
 libwxsmithlib_la_LIBADD = $(PLUGIN_WX_LIBS) \
 				../../../sdk/libcodeblocks.la \
--- origsrc/codeblocks-12.11release8629/src/sdk/Makefile.am	2013-10-09 19:19:13.100324600 -0500
+++ src/codeblocks-12.11release8629/src/sdk/Makefile.am	2013-10-09 14:06:00.000000000 -0500
@@ -12,7 +12,7 @@ INCLUDES = $(WX_CXXFLAGS) \
 
 lib_LTLIBRARIES = libcodeblocks.la
 
-libcodeblocks_la_LDFLAGS = -version-info @CODEBLOCKS_VERSION_INFO@
+libcodeblocks_la_LDFLAGS = -version-info @CODEBLOCKS_VERSION_INFO@ -no-undefined
 
 libcodeblocks_la_LIBADD =  $(WX_LIBS) \
 			scripting/bindings/libsqbindings.la \
--- origsrc/codeblocks-12.11release8629/src/sdk/pluginmanager.cpp	2013-10-09 19:19:13.106324900 -0500
+++ src/codeblocks-12.11release8629/src/sdk/pluginmanager.cpp	2013-10-09 14:06:00.000000000 -0500
@@ -276,11 +276,11 @@ bool PluginManager::InstallPlugin(const
     wxString resourceName = basename + _T(".zip");
     wxString settingsOnName = basename + _T(".png");
     wxString settingsOffName = basename + _T("-off.png");
-    if (!platform::windows && resourceName.StartsWith(_T("lib")))
+    if (!platform::windows && resourceName.StartsWith(_T("cyg")))
         resourceName.Remove(0, 3);
-    if (!platform::windows && settingsOnName.StartsWith(_T("lib")))
+    if (!platform::windows && settingsOnName.StartsWith(_T("cyg")))
         settingsOnName.Remove(0, 3);
-    if (!platform::windows && settingsOffName.StartsWith(_T("lib")))
+    if (!platform::windows && settingsOffName.StartsWith(_T("cyg")))
         settingsOffName.Remove(0, 3);
     wxString pluginFilename = pluginDir + _T('/') + localName;
 //    Manager::Get()->GetLogManager()->DebugLog(F(_T("Plugin filename: ") + pluginFilename));
@@ -382,11 +382,11 @@ bool PluginManager::UninstallPlugin(cbPl
             resourceFilename = fname.GetName() + _T(".zip");
             settingsOnFilename = fname.GetName() + _T(".png");
             settingsOffFilename = fname.GetName() + _T("-off.png");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
-            if (!platform::windows && settingsOnFilename.StartsWith(_T("lib")))
+            if (!platform::windows && settingsOnFilename.StartsWith(_T("cyg")))
                 settingsOnFilename.Remove(0, 3);
-            if (!platform::windows && settingsOffFilename.StartsWith(_T("lib")))
+            if (!platform::windows && settingsOffFilename.StartsWith(_T("cyg")))
                 settingsOffFilename.Remove(0, 3);
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
             settingsOnFilename = ConfigManager::LocateDataFile(_T("images/settings/") + settingsOnFilename, sdDataGlobal | sdDataUser);
@@ -517,14 +517,14 @@ bool PluginManager::ExportPlugin(cbPlugi
 
             // now get the resource zip filename
             resourceFilename = fname.GetName() + _T(".zip");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
             sourcefiles.Add(resourceFilename);
 
             // the highlighted icon the plugin may have for its "settings" page
             resourceFilename = fname.GetName() + _T(".png");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             resourceFilename.Prepend(_T("images/settings/"));
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
@@ -533,7 +533,7 @@ bool PluginManager::ExportPlugin(cbPlugi
 
             // the non-highlighted icon the plugin may have for its "settings" page
             resourceFilename = fname.GetName() + _T("-off.png");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             resourceFilename.Prepend(_T("images/settings/"));
             resourceFilename = ConfigManager::LocateDataFile(resourceFilename, sdDataGlobal | sdDataUser);
@@ -542,7 +542,7 @@ bool PluginManager::ExportPlugin(cbPlugi
 
             // export extra files
             resourceFilename = fname.GetName() + _T(".zip");
-            if (!platform::windows && resourceFilename.StartsWith(_T("lib")))
+            if (!platform::windows && resourceFilename.StartsWith(_T("cyg")))
                 resourceFilename.Remove(0, 3);
             ReadExtraFilesFromManifestFile(resourceFilename, extrafilesdest);
             for (size_t n = 0; n < extrafilesdest.GetCount(); ++n)
@@ -730,7 +730,7 @@ bool PluginManager::ReadManifestFile(con
         wxString actual = fname.GetFullName();
 
         // remove 'lib' prefix from plugin name (if any)
-        if (!platform::windows && actual.StartsWith(_T("lib")))
+        if (!platform::windows && actual.StartsWith(_T("cyg")))
             actual.Remove(0, 3);
 
         actual = ConfigManager::LocateDataFile(actual, sdPluginsUser | sdDataUser | sdPluginsGlobal | sdDataGlobal);
@@ -870,7 +870,7 @@ void PluginManager::ReadExtraFilesFromMa
     wxString actual = fname.GetFullName();
 
     // remove 'lib' prefix from plugin name (if any)
-    if (!platform::windows && actual.StartsWith(_T("lib")))
+    if (!platform::windows && actual.StartsWith(_T("cyg")))
         actual.Remove(0, 3);
 
     actual = ConfigManager::LocateDataFile(actual, sdPluginsUser | sdDataUser | sdPluginsGlobal | sdDataGlobal);
@@ -928,7 +928,7 @@ void PluginManager::ReadExtraFilesFromMa
 
 int PluginManager::ScanForPlugins(const wxString& path)
 {
-    static const wxString PluginsMask = platform::windows ? _T("*.dll") : _T("*.so");
+    static const wxString PluginsMask = _T("*.dll");
     int count = 0;
     if(!wxDirExists(path))
         return count;
@@ -946,10 +946,7 @@ int PluginManager::ScanForPlugins(const
         if (!bbplugins.GetCount())
         {
             // defaults
-            if(platform::windows)
-                bbplugins.Add(_T("compiler.dll"));
-            else
-                bbplugins.Add(_T("libcompiler.so"));
+            bbplugins.Add(_T("cygcompiler.dll"));
         }
     }
 
--- origsrc/codeblocks-12.11release8629/src/src/compilersettingsdlg.cpp	2013-10-09 19:19:13.109325100 -0500
+++ src/codeblocks-12.11release8629/src/src/compilersettingsdlg.cpp	2013-10-09 14:06:00.000000000 -0500
@@ -66,10 +66,7 @@ CompilerSettingsDlg::CompilerSettingsDlg
     if (!bbplugins.GetCount())
     {
         // defaults
-        if(platform::windows)
-            bbplugins.Add(_T("compiler.dll"));
-        else
-            bbplugins.Add(_T("libcompiler.so"));
+        bbplugins.Add(_T("cygcompiler.dll"));
     }
     wxCheckListBox* clb = XRCCTRL(*this, "chkBBPlugins", wxCheckListBox);
     clb->Clear();
@@ -241,7 +238,7 @@ void CompilerSettingsDlg::EndModal(int r
             }
         }
 
-        const wxString compiler(platform::windows ? _T("compiler.dll") : _T("libcompiler.so"));
+        const wxString compiler(_T("cygcompiler.dll"));
 
         if (bbplugins.Index(compiler) == wxNOT_FOUND)
         {
