glib-2.22 adds a GSocket class which conflicts with wxGTK

--- origsrc/codeblocks-8.02/src/plugins/contrib/codesnippets/codesnippets.cpp	2009-12-16 04:16:23.406919400 -0600
+++ src/codeblocks-8.02/src/plugins/contrib/codesnippets/codesnippets.cpp	2009-12-16 04:14:31.570522800 -0600
@@ -52,8 +52,10 @@
 #include "memorymappedfile.h"
 
 #if defined(__WXGTK__)
+    #define GSocket GLibSocket
     #include "wx/gtk/win_gtk.h"
     #include <gdk/gdkx.h>
+    #undef GSocket
 #endif
 
 // The plugin needs a flag ON to enable some code for the plugin
--- origsrc/codeblocks-8.02/src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp	2009-12-16 04:16:23.508925300 -0600
+++ src/codeblocks-8.02/src/plugins/contrib/codesnippets/codesnippetstreectrl.cpp	2009-12-16 04:14:42.501148000 -0600
@@ -49,8 +49,10 @@
 #include "menuidentifiers.h"
 #include "editsnippetframe.h"
 #if defined(__WXGTK__)
+    #define GSocket GLibSocket
     #include "wx/gtk/win_gtk.h"
     #include <gdk/gdkx.h>
+    #undef GSocket
 #endif
 
 IMPLEMENT_DYNAMIC_CLASS(CodeSnippetsTreeCtrl, wxTreeCtrl)
