Causes segfault on exit.

--- origsrc/codeblocks-8.02/src/plugins/contrib/byogames/byogame.cpp	2006-06-21 17:00:16.000000000 -0500
+++ src/codeblocks-8.02/src/plugins/contrib/byogames/byogame.cpp	2009-12-16 04:29:19.434305700 -0600
@@ -20,7 +20,7 @@ byoGameLauncher::byoGameLauncher(const w
 
 byoGameLauncher::~byoGameLauncher()
 {
-    GetGames().Remove(this);
+//    GetGames().Remove(this);
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////
